### PR TITLE
[17.0][MIG] sale_order_carrier_auto_assign: Migration to 17.0

### PR DESCRIPTION
**Pull request by lef-adhoc**

### CODE REVIEW

**Comment by rousseldenis** [comment]
/ocabot migration sale_order_carrier_auto_assign

@lef-adhoc Could you rebase ?

**Comment by lef-adhoc** [comment]
@rousseldenis the rebase is ready

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by FrancoMaxime — APPROVED** [review]
LGTM: code review

File: sale_order_carrier_auto_assign/tests/test_sale_order_carrier_auto_assign.py [hunk 1]
```diff
@@ -32,7 +33,9 @@ def test_sale_order_carrier_auto_assign(self):
         self.sale_order.action_confirm()
         self.assertEqual(self.sale_order.state, "sale")
         self.assertEqual(self.sale_order.carrier_id, self.delivery_local_delivery)
-        delivery_line = self.sale_order.order_line.filtered(lambda l: l.is_delivery)
+        delivery_line = self.sale_order.order_line.filtered(
```
Comment (rousseldenis) [inline sale_order_carrier_auto_assign/tests/test_sale_order_carrier_auto_assign.py:36]: FYI, `self.sale_order.order_line.filtered("is_delivery")` is working well.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**